src/curate_gpt/agents/base_agent.py

```diff
@@ -26,3 +26,7 @@ class BaseAgent(ABC):
 
     extractor: Extractor = None
     """Engine performing LLM operations, including extracting from prompt responses"""
+
+    @abstractmethod
+    def search(self):
```

Comment:
not every agent implements search

Reply:
Ah, I removed that from the BaseWrapper but not the BaseAgent, apologies

Reply:
*that* being abstract methods